src/model/Model.lua

```diff
@@ -168,7 +168,33 @@ end
      * @return the model text.
 ]]
 function Model:saveModelToText()
-
+    local tokenPatterns={}
+    for _,ptype in pairs({"r","p"}) do
+        for _,token in pairs(self.model[ptype][ptype].tokens) do
+            tokenPatterns[token]=string.gsub (string.gsub (token,"^p_","p."),"^r_","r.")
+        end
+    end
+    local s=""
+    local writeString=function(sec)
+        local result=""
+		for ptype,_ in pairs(self.model[sec]) do
+            local value=self.model[sec][ptype].value
+            for tokenPattern,newToken in pairs(tokenPatterns) do
+                value=string.gsub(value,tokenPattern,newToken)
+            end
+            s=s..sec.."="..value.."\n"
```

Comment:
`result = result .. ptype.."="..self.model["g"][ptype].value.."\n"`

Reply:
@Edmond-J-A replace s with result instead.

Reply:
Solved，I forgot to change this.